pandas/_libs/interval.pyx

```diff
@@ -359,6 +360,35 @@ cdef class Interval(IntervalMixin):
                 self.left // y, self.right // y, closed=self.closed)
         return NotImplemented
 
+    def overlaps(self, other):
+        """
+        Check whether two interval objects overlap.
+
+        .. versionadded:: 0.24.0
+
+        Parameters
+        ----------
+        other : Interval
+            The interval to check against for an overlap.
+
+        Returns
+        -------
+        bool
+            ``True`` if the two intervals overlap, else ``False``.
```

Comment:
We should clarify what "overlap" exactly means regarding the boundaries of the interval

Reply:
done (further up in the docstring)